resources/js/components/ui/calendar/TinyCalendar.tsx

```diff
@@ -0,0 +1,216 @@
+    if (lProps.month.hasSame(currentDate(), "month")) {
+      const selCenter = selectionCenter();
+      if (!selCenter) {
+        return undefined;
+      }
+      return lProps.month.hasSame(selCenter, "month") ? undefined : "toSelection";
+    } else {
+      return "toCurrent";
+    }
+  });
+
+  /** List of days to show in the calendar. */
+  const days = createMemo(() => {
+    const holidaysSet = new Set(lProps.holidays?.map((d) => d.startOf("day").toMillis()));
+    let day = weekDaysCalculator().startOfWeek(monthStart().minus({days: lProps.minDaysOfPrevMonth}));
+    const res: DayInfo[] = [];
+    function add(day: DateTime) {
+      const isToday = day.hasSame(currentDate(), "day");
+      res.push({
+        day,
+        isToday,
+        classes: cx({
+          [s.today!]: isToday,
+          [s.weekend!]: weekInfo().weekend.includes(day.weekday),
+          [s.holiday!]: holidaysSet.has(day.toMillis()),
+          [s.otherMonth!]: day.month !== monthStart().month,
+        }),
+      });
+    }
+    const last = monthStart().endOf("month").plus({days: lProps.minDaysOfNextMonth});
+    while (day <= last) {
+      add(day);
+      day = day.plus({days: 1});
+    }
+    while (res.length < 7 * lProps.minWeeks || day.weekday !== weekInfo().firstDay) {
+      add(day);
+      day = day.plus({days: 1});
+    }
+    return res;
+  });
+
+  /**
+   * Returns a classlist based on the current day and selection. Sets the specified class for days
+   * in the range, and additionally classes start and end on the first and last day of the range.
+   */
+  function rangeClasses(day: DateTime, range: DaysRange | undefined, rangeClass: string | undefined) {
+    return range && rangeClass
+      ? {
+          [rangeClass]: day >= range![0] && day <= range![1],
+          [s.start!]: day.hasSame(range![0], "day"),
+          [s.end!]: day.hasSame(range![1], "day"),
+        }
+      : undefined;
+  }
+
+  return (
+    <div {...htmlAttributes.merge(divProps, {class: s.tinyCalendar})}>
+      <div class={s.header}>
+        <button onClick={() => lProps.setMonth(lProps.month.minus({months: 1}))}>
+          <FaSolidArrowLeft />
+        </button>
+        <button onClick={() => lProps.setMonth(lProps.month.plus({months: 1}))}>
+          <FaSolidArrowRight />
+        </button>
+        <div class={s.date}>
+          <div>{lProps.month.monthLong}</div>
+          <div>{lProps.month.year}</div>
+        </div>
+        <button
+          disabled={!retButtonAction()}
+          onClick={() => lProps.setMonth((retButtonAction() === "toSelection" && selectionCenter()) || currentDate())}
+          title={
+            retButtonAction() === "toSelection" ? t("calendar.go_to_selection") : t("calendar.go_to_current_month")
+          }
+        >
+          <CgCalendarToday classList={{dimmed: !retButtonAction()}} />
+        </button>
+      </div>
+      <div class={s.days}>
+        <Show when={lProps.showWeekdayNames}>
```

Comment:
Będzie zapotrzebowanie na showWeekdayNames=false? Bo mam wątpliwości do takich martwych propsów.

Reply:
No niech sobie będzie jak musi być, no ale kod też można zmieniać, więc niezdecydowanie nie usprawiedliwia dla mnie propsa. :) Propsa to widzę jakby raz było tak a raz tak. A jak wszędzie tak samo, ale jeszcze nie wiadomo jak to lepiej wg mnie zrobić jedną z wersji, a zawsze można zmienić.

Reply:
Ale którą wersję?! A poza tym mówiłem, wolę napisać obie teraz gdy wiem jak to działa niż potem musieć to znowu zrozumieć. Chcę zrobić dobrego black boxa w miarę możliwości.